infra/pr_helper.py

```diff
@@ -0,0 +1,219 @@
+  """Retrieves the author's email address for a pull request,
+  including non-public emails."""
+  commits_response = requests.get(f'{BASE_URL}/pulls/{pr_number}/commits',
+                                  headers=headers)
+  if not commits_response.ok:
+    return None
+  email = commits_response.json()[0]['commit']['author']['email']
+  return email
+
+
+def get_project_yaml(project_path, headers):
+  """Gets the project yaml file."""
+  contents_url = f'{BASE_URL}/contents/{project_path}/project.yaml'
+  return get_yaml_file_content(contents_url, headers)
+
+
+def get_yaml_file_content(contents_url, headers):
+  """Gets yaml file content."""
+  response = requests.get(contents_url, headers=headers)
+  if not response.ok:
+    return {}
+  content = base64.b64decode(response.json()['content']).decode('UTF-8')
+  return yaml.safe_load(content)
+
+
+def get_integrated_project_info(pr_number, headers):
+  """Gets the new integrated project."""
+  response = requests.get(f'{BASE_URL}/pulls/{pr_number}/files',
+                          headers=headers)
+
+  for file in response.json():
+    file_path = file['filename']
+    if 'project.yaml' in file_path:
+      return get_yaml_file_content(file['contents_url'], headers)
+
+  return None
+
+
+def get_criticality_score(repo_url):
+  """Gets the criticality score of the project."""
+  report = subprocess.run(
+      ['criticality_score', '--format', 'json', '--repo', repo_url],
+      capture_output=True,
+      text=True)
+  content = report.stderr.split('\n')
+  scores = {}
+  for entry in content:
+    kv_entry = entry.split(': ')
+    if len(kv_entry) == 2:
+      key = kv_entry[0].strip().replace(',', '').replace('"', '')
+      value = kv_entry[1].strip().replace(',', '')
+      scores[key] = value
+
+  return scores.get('criticality_score', 'N/A')
+
+
+def is_known_contributor(content, email):
+  """Checks if the author is in the contact list."""
+  return (email == content.get('primary_contact') or
+          email in content.get('vendor_ccs', []) or
+          email in content.get('auto_ccs', []))
+
+
+def has_author_modified_project(project_path, pr_author, headers):
+  """Checks if the author has modified this project before."""
+  commits_response = requests.get(
+      f'{BASE_URL}/commits?path={project_path}&author={pr_author}',
+      headers=headers)
+
+  if not commits_response.ok or not commits_response.json():
+    return False
+
+  commit = commits_response.json()[0]
+  return True, commit['sha']
+
+
+def get_pull_request_url(commit, headers):
+  """Gets the pull request url."""
+  pr_response = requests.get(f'{BASE_URL}/commits/{commit}/pulls',
+                             headers=headers)
+  if not pr_response.ok:
+    return None
+  return pr_response.json()[0]['html_url']
+
+
+def is_author_internal_member(pr_author):
+  """Returns if the author is an internal member."""
+  internal_members = ['Alan32Liu', 'hogo6002', 'jonathanmetzman', 'oliverchang']
```

Comment:
Let's instead source this from a file in our repo, so it's more obvious to us where this is. e.g. a  MAINTAINERS file in the oss-fuzz repo root.